src/spikeinterface/extractors/nwbextractors.py

```diff
@@ -451,13 +402,378 @@ def __init__(
+            "electrical_series_name": self._electrical_series_name,
+            "load_time_vector": load_time_vector,
+            "samples_for_rate_estimation": samples_for_rate_estimation,
+            "stream_mode": stream_mode,
+            "cache": cache,
+            "stream_cache_path": stream_cache_path,
+            "file": file,
+            "electrical_series_location": electrical_series_location,
+        }
+
+    @staticmethod
+    def find_electrical_series(group, path="", result=None):
+        """
+        Recursively searches for groups with neurodata_type 'ElectricalSeries' in the HDF5 group or file,
+        and returns a dictionary with their names and paths.
+        """
+        import h5py
+
+        if result is None:
+            result = {}
+
+        for neurodata_name, value in group.items():
+            # Check if it's a group and if it has the 'ElectricalSeries' neurodata_type
+            if isinstance(value, h5py.Group):
+                current_path = f"{path}/{neurodata_name}" if path else neurodata_name
+                if value.attrs.get("neurodata_type") == "ElectricalSeries":
+                    result[neurodata_name] = current_path
+                _NWBHDF5RecordingExtractor.find_electrical_series(
+                    value, current_path, result
+                )  # Recursive call for sub-groups
+
+        return result
+
+
+class NwbRecordingExtractor(BaseRecording):
+    """Load an NWBFile as a RecordingExtractor.
+
+    Parameters
+    ----------
+    file_path: str, Path, or None
+        Path to the NWB file or an s3 URL. Use this parameter to specify the file location
+        if not using the `file` parameter.
+    electrical_series_name: str or None, default: None
+        The name of the ElectricalSeries object within the NWB file. This parameter is crucial
+        when the NWB file contains multiple ElectricalSeries objects. It helps in identifying
+        which specific series to extract data from. If there is only one ElectricalSeries and
+        this parameter is not set, that unique series will be used by default.
+    file: file-like object or None, default: None
+        A file-like object representing the NWB file. Use this parameter if you have an in-memory
+        representation of the NWB file instead of a file path.
+    load_time_vector: bool, default: False
+        If set to True, the time vector is also loaded into the recording object. Useful for
+        cases where precise timing information is required.
+    samples_for_rate_estimation: int, default: 100000
+        The number of timestamp samples used for estimating the sampling rate. This is relevant
+        when the 'rate' attribute is not available in the ElectricalSeries.
+    stream_mode : "fsspec", "ros3", "remfile", or None, default: None
+        Determines the streaming mode for reading the file. Use this for optimized reading from
+        different sources, such as local disk or remote servers.
```

Comment:
```suggestion
        different sources, such as local disk or remote servers. ("remfile" is recommended)
```

Reply:
I would not like to commit so fast to this yet:

https://github.com/NeurodataWithoutBorders/pynwb/issues/1791

I also seen too much variability around my own usage. I prefer to defer this until the nwb team give a recommendation.